parser-typechecker/src/Unison/Typechecker/Components.hs

```diff
@@ -43,7 +49,7 @@ minimize (Term.LetRecNamedAnnotatedTop' isTop ann bs e) =
   in  if not $ null dupes
         then Left $ Nel.fromList dupes
         else
-          let cs0            = ABT.components bindings
+          let cs0 = if isTop then unordered bindings else ordered bindings
```

Comment:
Top level definitions can be reordered - the order of definitions in a file isn't semantically meaningful, but the order of definitions within a function body might be so we conservatively don't do any reordering.